lib/query/where.js

```diff
@@ -53,46 +57,65 @@ var WhereBuilder = ASTVisitor.extend({
         }
     },
 
-    visitNumericLiteral: function(node) {
+    visitNumericLiteral: function(node, op, options) {
         return node.value;
     },
 
-    visitNullLiteral: function(node) {
+    visitNullLiteral: function(node, op, options) {
         return 'null';
     },
 
-    visitVariable: function(node) {
+    visitVariable: function(node, op, options) {
         return '"' + node.name + '"';
     },
 
-    visitUnaryExpression: function(node) {
+    visitUnaryExpression: function(node, op, options) {
         // Field dereference
         if (node.operator === '*') {
             return '"' + node.expression.value + '"';
         } else {
-            var op = this.ops[node.operator];
+            var operator = this.ops[node.operator];
 
-            if (!op) { throw new Error('Operator ' + node.operator + ' not supported by InfluxQL'); }
+            if (!operator) { throw new Error('Operator ' + node.operator + ' not supported by InfluxQL'); }
 
-            return '( ' + op + ' ' + this.visit(node.expression) + ' )';
+            return '( ' + operator + ' ' + this.visit(node.expression, operator, options) + ' )';
         }
     },
 
-    visitBinaryExpression: function(node) {
-        var op = this.ops[node.operator];
+    isNameField: function(node, nameField) {
```

Comment:
Since you have 3-4 cases in which you implement the same `isNameField` semantics, I think would make sense to DRY this up a bit and add a common base class that you use within this adapter.